guides/advancing-proofs/simplifications-guide.md

```diff
@@ -98,7 +103,157 @@ to which `div-group-conc` would then apply, but not `div-group-conc-b`.
 
 Second, note that this simplification has the `preserves-definedness` attribute. This attribute should be used when either the LHS or the RHS contains a **partial function**, but we are certain that both the LHS and RHS are defined for all use cases. In this case, the partial function is `/Int` and both the LHS and the RHS are guaranteed to be defined given the `notBool C ==Int 0` requirement.
 
-## Creating advanced simplifications: automating storage slot updates
+
+
+
+## More advanced simplifications: Handling the tension between byte arrays and integers
+
+In EVM, data can be stored in two ways---as 32-byte (or 256-bit) integers (for example, in the word stack or in storage) or as byte arrays (for example, in the memory)---and the language semantics often needs to switch between these views and manipulate the associated data. The KEVM functions relevant for this part of the symbolic reasoning of Kontrol are:
+
+- `#asWord(B:Bytes) : Bytes -> Int`, which denotes the unsigned integer represented by the lowest 32 bytes (or 256 bits) of byte array `B`. This function is total.
+- `#buf(WIDTH:Int, VALUE:Int) : Int -> Int -> Bytes`, which denotes a byte array (or buffer) `WIDTH` bytes long, the contents of which, when interpreted as an *unsigned integer*, equal `VALUE`. For example, `#buf(1, 9)` represents a single byte with contents `00010001`. For `#buf` to be well-defined, the following conditions need to be met:
```

Comment:
Shouldn't `#buf(1,9)` be `00001001` instead of `00010001`?

Reply:
I think it's `00001001` too, will edit.

Reply:
Sorry, Petar already did.